### ISSUE TITLE
Group of Ikea CT Bulbs. Always some hung bulbs after a few days.

### ISSUE THREAD

**Issue by olemr** [dresden-elektronik/deconz-rest-plugin#1086]
Currently on 
![image](https://user-images.githubusercontent.com/28068256/50709644-c3417180-1068-11e9-810c-51c2a18a3ee9.png)
,but have had the same issue since late summer.

I have a total of 33 Ikea CT bulbs. 
16 of them are E27 facade lights grouped to an Ikea remote (round 5 button). 8 fixtures with 2 bulbs in each.
12 of them are GU10 in two 6 bulbs fixtures in the same room. One Ikea remote (5b) grouped to each fixture, one deconz group containing all 12.
2 E27 are in a window
3 others are single groups, one bulb in each group.

What I'm experiencing is that in the group of 16 bulbs only 1-2 days pass before at least one bulb hangs. Soon more follow. Even the remote cannot control them, and they have to be power cycled.
In the group of 12, more days passes before bulbs die.

In the group of 2, I have only experienced 1 hang in ~6mnds.
In the 1 bulb groups, I have not had any.

This leads me to ask the question: How does deconz handle group commands? Could there be some marginal timing causing the bulbs to hang when issuing commands to larger groups?

All bulbs are on "swversion": "1.2.217"

**Comment by olemr** [comment]
I'm now on .54 and I understand Bulb polling has been reduced.
I also had a review of my own control code using the deconz REST API, and found a place where I had only 50ms between 'on' and 'bri'.
For my two largest group of lights I now have 10s between REST API commands, and it will be very interesting to watch how that goes in the next couple of days.

**Comment by manup** [comment]
Thanks for looking into this, please report back if it behaves better now.

**Comment by olemr** [comment]
This is looking really good. Not even **one** bulb greyed out :-)
(This has never happened before. Group commands worked for a while even though some lights were grey.)
![image](https://user-images.githubusercontent.com/28068256/50996328-de592900-1521-11e9-9b9b-6c7f16f92423.png)

**Comment by manup** [comment]
Nice, hope it stays stable :)

**Comment by olemr** [comment]
All lights still reachable. This looks very promising.
Problem was probably that I sent 'on' & 'bri' too close in time.
What the actual limit is I don't know, but I'm currently spacing all commands by 1500ms.
I feel 1 of 2 things needs to be done.
1. Honour the REST API documentation that states that commands can be grouped in one command. This means that deconz itself must på Bulb vendor aware.
2. if 1. not feasible, at least list known restrictions in the Doc.

Now I'm really considering pairing my remaining 28 Trådfri devices still on the Ikea HUB.
The Ikea HUB has no way of saving the configuration/pairing info, and I lost it all here the other day.
Also openHAB integration with the HUB is a bit flaky (when restarting the system, fine when up).

**Comment by manup** [comment]
Note when you put a `{ "on": true, "bri": 5 }` it will result in only one Zigbee message (Move to Level with OnOff).

For Ikea it is important to not interfere commands while transition time (default 400ms) of a former command is active.

I don't remember correctly about the details but there were also commands (with bri, ct) which could nock out Ikea bulbs so that a power-cycle was needed. When we have a clearer picture we can describe this in the documentation as Ikea specifics. Hopefully they will also fix these issues in the firmware.

Handling this in deCONZ might be possible but I would rather avoid this.

**Comment by ebaauw** [comment]
> Handling this in deCONZ might be possible but I would rather avoid this.

I fear we might eventually need to mimic the commands the various bridges/hubs/gateways send to "their" lights.  I already did that for `"alert": "select"` (the Hue bridge issues a _Breathe_ _Effect_ instead of an _Indentify_ command).  Adding a default transitiontime of 0.4 seconds is definitely a Hue thing.  I also saw one issue, where I think a GLEDOPTO doesn't handle _Move to Level with OnOff_, see https://github.com/ebaauw/homebridge-hue/issues/424.
